pkg/handlers/generic/mutation/extraapiservercertsans/inject.go

```diff
@@ -63,24 +65,35 @@ func (h *extraAPIServerCertSANsPatchHandler) Mutate(
 		h.variableFieldPath...,
 	)
 	if err != nil {
-		return err
+		log.Error(
+			err,
+			"failed to get cluster config variable from extraAPIServerCertSANs mutation handler",
+		)
 	}
 	if !found {
 		log.V(5).Info("Extra API server cert SANs variable not defined")
 	}
-	apiCertSANs := extraAPIServerCertSANsVar
+	cluster, err := clusterGetter(ctx)
+	if err != nil {
+		log.Error(
+			err,
+			"failed to get cluster from extraAPIServerCertSANs mutation handler",
+		)
+	}
+	defaultAPICertSANs := getDefaultAPIServerSANs(cluster)
+	//nolint: gocritic // this is more readable
+	apiCertSANs := append(extraAPIServerCertSANsVar, defaultAPICertSANs...)
```

Comment:
nit: maybe we can avoid ignoring this linter by a little refactor https://github.com/go-critic/go-critic/issues/865#issuecomment-522964363

```
func combinedAPICertSANsForCluster(ctx context.Context, clusterGetter mutation.ClusterGetter, extraAPIServerCertSANs []string) ([]string, error) {
	cluster, err := clusterGetter(ctx)
	if err != nil {
		return nil, fmt.Errorf("failed to get cluster from extraAPIServerCertSANs mutation handler: %w", err)
	}
	defaultAPICertSANs := getDefaultAPIServerSANs(cluster)

	return append(extraAPIServerCertSANs, defaultAPICertSANs...), nil
}
```

```suggestion
	apiCertSANs, err := combinedAPICertSANsForCluster(ctx, clusterGetter, extraAPIServerCertSANsVar)
	if err != nil {
		log.Error(
			err,
			"failed to get SANs for API server",
		)
	}
```

Reply:
nice find. lets see what i can do here.

Reply:
i think using a copy would be cleanest approach